### PR TITLE
Fixes #10842: Update write test for 4.1 and add generation for a node

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/10842

### CODE REVIEW

File: rudder-core/src/test/scala/com/normation/rudder/services/policies/write/WriteSystemTechniquesTest.scala [hunk 1]
```diff
@@ -207,31 +194,48 @@ class WriteSystemTechniquesTest extends Specification with Loggable with BoxSpec
...
+
+    //where the "/var/rudder/share" file is for tests:
+    val SHARE = abstractRoot/s"share-${label}"
+
+    val rootGeneratedPromisesDir = SHARE/"root"
+
+    val pathComputer = new PathComputerImpl(
+        SHARE.getParent + "/"
+      , SHARE.getName
+      , abstractRoot.getAbsolutePath + "/backup"
```
Comment (VinceMacBuche) [inline rudder-core/src/test/scala/com/normation/rudder/services/policies/write/WriteSystemTechniquesTest.scala]: no prob if backup is the same for different label ?

File: rudder-core/src/test/scala/com/normation/rudder/services/policies/write/WriteSystemTechniquesTest.scala [hunk 2]
```diff
@@ -207,31 +194,48 @@ class WriteSystemTechniquesTest extends Specification with Loggable with BoxSpec
...
+
+    (rootGeneratedPromisesDir, promiseWritter)
+  }
 
   //////////// end init ////////////
 
   //////////// set-up auto test cleaning ////////////
   override def afterAll(): Unit = {
-    if(System.getProperty("tests.clean.tmp") != "false") {
+    if(false) { // System.getProperty("tests.clean.tmp") != "false") {
```
Comment (VinceMacBuche) [inline rudder-core/src/test/scala/com/normation/rudder/services/policies/write/WriteSystemTechniquesTest.scala]: Remove dead code (or set back condition)

**Comment by fanf** [comment]
PR rebased

File: rudder-core/src/test/scala/com/normation/rudder/services/policies/write/WriteSystemTechniquesTest.scala (hunk 1 above)
Comment (fanf) [inline rudder-core/src/test/scala/com/normation/rudder/services/policies/write/WriteSystemTechniquesTest.scala]: it doesn't seems to care. And at least for the generation part, it does not seems to matter.

File: rudder-core/src/test/scala/com/normation/rudder/services/policies/write/WriteSystemTechniquesTest.scala (hunk 2 above)
Comment (fanf) [inline rudder-core/src/test/scala/com/normation/rudder/services/policies/write/WriteSystemTechniquesTest.scala]: done

**Comment by VinceMacBuche** [comment]
I see noihing strange or invaluid apart the small remarks, merging

**Pull request merged**